File: lib/statsailr_procs_base/proc_setting/proc_mult.rb
module ProcMult
  include ProcSettingModule

  source_r_file(__dir__, File.basename(__FILE__ , ".rb") + ".R")
  validate_option("data", is_a: ["SymbolR", "String"], as: "SymbolR" , required: true)

  def setting_for_aov( setting )
    setting.libname = "stats"
    setting.func_name = "aov"
    setting.main_arg_and_how_to_treat = [ "formula" , :read_as_formula, :no_nil]
    setting.runtime_args = {"data" => param("data")}
    setting.store_result = true
    setting.print_opt = "summary"
  end

  def setting_for_tukey( setting )
    setting.libname = "stats"
    setting.func_name = "TukeyHSD"
    setting.main_arg_and_how_to_treat = nil
    setting.runtime_args = {"x" => result("aov")}
    setting.store_result = false
    setting.print_opt = true
  end

  def setting_for_p_adjust( setting )
    setting.libname = nil
    setting.envname = "sts_mult"
    setting.func_name = "p_adjust"
    setting.main_arg_and_how_to_treat = [ "method" , :read_as_strvec, :no_nil]
    setting.runtime_args = {"x" => result("aov")}
    setting.store_result = true
    setting.print_opt = true
  end
end

File: lib/statsailr_procs_base/proc_setting/proc_mult.R
STS_RC_Mult = setRefClass("STS_RC_Mult")
STS_RC_Mult$methods(
  p_adjust = function( x , method ){
    ori_p = summary(x)[[1]][['Pr(>F)']]
    result = p.adjust( ori_p, method )

    cat( "p adjustment\n" )
    cat( paste( "Method:", method, "\n", sep=" ", collapse=" " ) )
    return(result)
  }
)

sts_mult = STS_RC_Mult()
